### DISCUSSION TITLE
How to set status visible for certain roles

### DISCUSSION THREAD

**Discussion by rynskiy** [directus/directus#15086] [Archive]
Hi,

I'm new by directus and can't find any docs to set a collection status visible only for certain roles. 

Detailed: I have a collection "Articles". This collection has field "Status" with values: Draft, Translate, Edit, Publish. I need to set a role "Writer" who can see in dropdown with Statuses by Article details, only two of them - "Translate" and "Edit". ("Draft" and "Publish" must be for role "Writer" invisible.

**Comment by azrikahar** [comment]
You should be able to control that by adding a custom Read permission to Writer role on collection "articles" 👍 Here's a quick clip on the custom read permission steps:

https://user-images.githubusercontent.com/42867097/184870113-08f138d6-c2f2-4699-9cc7-6d14f90e6d3e.mp4

The above setup should satisfy the `Writer can only see Articles with Translate or Edit status` (here we use `or` since a status can be both at the same time, at least in logical sense) requirement. For Field permissions, I've checked all of it for demo purpose, but you can also omit some of the fields (except id since we'll need primary key to actually read items) if you wish to limit which fields can be read by Writers.

Further reference in docs: https://docs.directus.io/configuration/users-roles-permissions.html#configure-permissions

**Reply by rynskiy** [reply]
Thanks [azrikahar](https://github.com/azrikahar)! You have right - now "Writer" can see by "Content" section only Articles with Status "To Rewrite" and "Rewriting"

But if "Writer" start to editing Article - he can still see all existing Statuses (see screenshot). And I need to show "Writer" in this dropdown only two of them ("To Rewrite" and "Rewriting"). How can I set this?
![Unbenannt2](https://user-images.githubusercontent.com/54798086/184884659-a407931b-a580-4657-b9a2-0e9a7c7d238a.PNG)

**Reply by azrikahar** [reply]
That's definitely an interesting and valid use case, but I believe we can't limit the items based on user or role yet. This does sound similar to #14828 so it might be possible once that gets implemented.

In the meantime, you should be able to at least use Field Validation on Writer role's update permission to make sure they can only update to some values, and they'll receive an error if they attempt the other values:

![chrome_6QCwighsrP](https://user-images.githubusercontent.com/42867097/184887068-41a257af-c85f-44ec-b891-e3a2e5699bbf.png)

**Reply by rynskiy** [reply]
Thanks [azrikahar](https://github.com/azrikahar). I have found another way to hide some Statuses for certain roles. I have used not a standard Field "Status", but have make own Collection with Statuses. And now i can hide a some Statuses for role "Writer" by Roles & Permissions.
